core/src/main/java/io/questdb/cairo/map/FastMap.java

```diff
@@ -333,22 +333,63 @@ private FastMapValue asNew(BaseKey keyWriter, int index, int hashCode, FastMapVa
         kPos = (kPos + 7) & ~0x7;
         setOffset(offsets, index, keyWriter.startAddress - heapStart);
         setHashCode(offsets, index, hashCode);
-        if (--free == 0) {
-            rehash();
-        }
         size++;
         return valueOf(keyWriter.startAddress, keyWriter.appendAddress, true, value);
     }
 
-    private FastMapValue probe0(BaseKey keyWriter, int index, int hashCode, int keySize, FastMapValue value) {
-        long offset;
-        while ((offset = getOffset(offsets, index = (++index & mask))) > -1) {
-            if (hashCode == getHashCode(offsets, index) && keyWriter.eq(offset)) {
-                long startAddress = heapStart + offset;
+    private int getProbeSequenceLength(int hash, int actualIndex) {
```

Comment:
Would be great to update the javadoc. It still has the following:
```
The offset list is used as a hash table with linear probing.
```

Reply:
@puzpuzpuz what do you think now?

Reply:
Looks good! Marking this one as resolved.

> The offset list is used as an open addressing hash map with Robin Hood probing.

The term to Google for is "Robin Hood hashing" (while, to me, it's a variation of linear probing), so might be nice to use it here.